model/interactive/src/main/proto/beam_interactive_api.proto

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Protocol Buffers describing a service that can be used in conjunction with
+ * the TestStream class in order to control a pipeline remotely.
+ */
+
+syntax = "proto3";
+
+package org.apache.beam.model.interactive.v1;
+
+option go_package = "interactive_v1";
+option java_package = "org.apache.beam.model.interactive.v1";
+option java_outer_classname = "BeamInteractiveApi";
+
+import "beam_runner_api.proto";
+import "google/protobuf/timestamp.proto";
+
+
+service InteractiveService {
```

Comment:
I would rather not, this is pretty specific to the InteractiveRunner.

Reply:
Sure, I'll move it to the beam_runner_api.proto.

Reply:
I would have suggested to make a new transform definition called ExternalTestStream (or some similar name) which is independent of this TestStream.

In general I believe it would be best if transforms had a narrow scope as the cost of choosing a new URN and to have a specific payload is meant to be inexpensive. The payload can be arbitrarily chosen and does not necessarily need to be in proto.

Now for a runner to support TestStream it now has to say what parts of the TestStream specification it supports which is a hassle to deal with. Also this leads to others following the same pattern since adding a new field to an existing transform may be backwards compatible from a decoding perspective but isn't backwards compatible if it changes the intent of the transform since a runner may not know of that field.